server/api/views/workspace.py

```diff
@@ -14,21 +15,135 @@ class CreateWorkspace(APIView):
     def post(self, request):
         response = {"error": {}}
 
-        # TODO: get workspace name from request with serializer once name implemented
+        # get name from request using serializer
+        serializer = WorkspaceSerializer(data=request.data)
+        if not serializer.is_valid():
+            response["error"]["code"] = 400
+            response["error"]["message"] = "Invalid request data"
+            return Response(response, status=status.HTTP_400_BAD_REQUEST)
 
+        # create new workspace
         workspace = Workspace.objects.create(
             created_by_id = request.user,
             owner_id = request.user
         )
 
+        # set name if present
+        if not (serializer.data.get('name',None) == None):
+            name = serializer.validated_data['name']
+            workspace.name = name
+            workspace.save()
+
+        # add creator of workspace as owner of the new workspace
         workspace_member = WorkspaceMember.objects.create(
             workspace_id = workspace,
             user_id = request.user,
             added_by_id = request.user
         )
 
+        # create permissions for owner
+        permissions  = MemberPermissions.objects.create(
+            workspace_id = workspace,
+            member_id = workspace_member,
+            IS_OWNER = True,
+            MANAGE_WORKSPACE_MEMBERS = True,
+            MANAGE_WORKSPACE_ROLES = True,
+            MANAGE_SCHEDULES = True,
+            MANAGE_TIME_OFF = True,
+        )
+
         return Response(response, status=status.HTTP_201_CREATED)
     
+class ModifyWorkspace(APIView): # change name or owner, can only be done by owner.
+    authentication_classes = [JWTAuthentication]
+    permission_classes = [IsAuthenticated]
+
+    def put(self, request):
+        '''
+        workspace_id
+        new_owner_id (optional)
+        name (optional)
+        '''
+
+        response = {"error": {}}
+
+        serializer = WorkspaceSerializer(data=request.data)
+        if not serializer.is_valid():
+            response["error"]["code"] = 400
+            response["error"]["message"] = "Invalid request data"
+            return Response(response, status=status.HTTP_400_BAD_REQUEST)
+        
+        # Verify body contains required fields
+        if "workspace_id" not in request.data:
```

Comment:
Why not use the workspace serializer to do this? If you include `workspace_id` and `owner_id` in the serializer, you can do all of this in one step and can set what is optional or not.